### PR TITLE
LPS-184524 GET and PUT by ERC SXPBlueprint

### PR DESCRIPTION
**Pull request by felipelorenz**
https://issues.liferay.com/browse/LPS-184524
Continued from here: https://github.com/liferay-search/liferay-portal/pull/951

### CODE REVIEW

**Comment by felipelorenz** [comment]
ci:test:relevant

File: modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java [hunk 1]
```diff
@@ -94,15 +94,18 @@ public SXPBlueprint getSXPBlueprintByExternalReferenceCode(
 			String externalReferenceCode)
 		throws Exception {
 
+		com.liferay.search.experiences.model.SXPBlueprint sxpBlueprint =
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java:97]: Is there a reason not to inline this, as it was?

**Comment by peerkar** [comment]
@felipelorenz looks good to me 👍  Just that one question. 

Having only 1 buildRest in a PR would be desirable, but if it's too much trouble, I think it's ok.

File: modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java (hunk 1 above)
Comment (felipelorenz) [reply modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java:97]: @peerkar My bad. I did this to change the parameter on the DefaultDTOConverterContext, I put externalReferenceCode, but its require an ID, I will fix this.

**Comment by felipelorenz** [comment]
ci:test:sf

File: modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java [hunk 2]
```diff
@@ -94,15 +94,18 @@ public SXPBlueprint getSXPBlueprintByExternalReferenceCode(
...
+			_sxpBlueprintService.getSXPBlueprintByExternalReferenceCode(
+				contextCompany.getCompanyId(), externalReferenceCode);
+
 		return _sxpBlueprintDTOConverter.toDTO(
 			new DefaultDTOConverterContext(
 				contextAcceptLanguage.isAcceptAllLanguages(), new HashMap<>(),
 				_dtoConverterRegistry, contextHttpServletRequest,
-				externalReferenceCode,
+				sxpBlueprint.getSXPBlueprintId(),
```
Comment (felipelorenz) [inline modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPBlueprintResourceImpl.java:105]: @peerkar Fixed! This is why we not inline. Its ok for you?

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by felipelorenz** [comment]
ci:test:relevant

**Comment by felipelorenz** [comment]
ci:test:relevant

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by brookedalton** [comment]
ci:test:search

**Comment by brookedalton** [comment]
ci:test:stable

**Comment by brookedalton** [comment]
ci:forward:force

**Pull request closed without merging**